### PR TITLE
[ADD] stock_measuring_device_zippcube

### PR DESCRIPTION
**Pull request by gurneyalex**
support for Bosche Zippcube measuring devices

includes #1121

### CODE REVIEW

**Comment by simahawk** [comment]
could you remove the PR #1121  and add a TMP dependency to the other branch?

File: stock_measuring_device_zippcube/components/zippcube_device_component.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+from odoo import _
+
+from odoo.addons.component.core import Component
+
+_logger = logging.getLogger(__name__)
+
+
+class ZippcubeDevice(Component):
+    _name = "zippcube.device.component"
```
Comment (simahawk) [inline stock_measuring_device_zippcube/components/zippcube_device_component.py]:
```suggestion
    _name = "measuring.device.zippcube"
```

File: stock_measuring_device_zippcube/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        device = env["measuring.device"].search(
+            [("name", "=", zippcube_device_name), ("device_type", "=", "zippcube")],
+            limit=1,
+        )
+        if not device:
+            raise MissingError(
+                _("No such Zippcube with name {}.").format(zippcube_device_name)
+            )
+
+        keys_missing = set(self.expected_keys) - set(data)
```
Comment (simahawk) [inline stock_measuring_device_zippcube/controllers/main.py:39]: A validator could save some coding/time here  eg: https://docs.python-cerberus.org/en/stable/schemas.html
Not a change request, of course :wink:

File: stock_measuring_device_zippcube/scripts/measurement.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+curl --header "Content-Type: application/json" \
```
Comment (simahawk) [inline stock_measuring_device_zippcube/scripts/measurement.sh:3]: this is only for sample I guess?

**Review by simahawk — APPROVED** [review]
overall LG

File: stock_measuring_device_zippcube/scripts/measurement.sh (hunk 3 above)
Comment (gurneyalex) [reply stock_measuring_device_zippcube/scripts/measurement.sh:3]: yes, useful for interactive tests, when you don't have the real thing at hand.

**Review by mmequignon — APPROVED** [review]

**Comment by simahawk** [comment]
@gurneyalex base PR merged, pls rebase

**Comment by simahawk** [comment]
@mmequignon please rebase and squash

**Pull request merged**